### PR TITLE
delete-multiple-cart-items

### PR DESCRIPTION
**Pull request by kriszhli**
Added an endpoint `delete-items`, so that multiple items can be deleted with a single request JSON that contains multiple values of product_id's with the key "product_ids".

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -47,20 +56,17 @@ jobs:
         run: |
           # stop the build if there are Python syntax errors or undefined names
           flake8 service tests --count --select=E9,F63,F7,F82 --show-source --statistics
-          # test for complexity. The GitHub editor is 127 chars wide
+          # check for omplexity. The GitHub editor is 127 chars wide
```
Comment (sahishnu) [inline .github/workflows/ci.yml:59]:
```suggestion
          # check for complexity. The GitHub editor is 127 chars wide
```

File: service/routes.py [hunk 2]
```diff
@@ -258,6 +258,37 @@ def delete_item(shopcart_id, item_id):
...
+
+    # Iterate through the list of product IDs and delete each product
+    for product_id in product_ids:
+        product = CartItem.find(product_id)
+        if product:
+            product.delete()
+        else:
+            # Return a 404 response if the product is not found
+            error_message = (
+                f"Product with ID {product_id} not found in shop cart {shopcart_id}"
```
Comment (sahishnu) [inline service/routes.py:285]: With `DELETE`, we always send back a `204_NO_CONTENT` even if the item is not found. So no need for this error response here.

File: tests/test_routes.py [hunk 3]
```diff
@@ -357,6 +358,76 @@ def test_delete_item_without_existing_item(self):
         )
         self.assertEqual(resp.status_code, status.HTTP_204_NO_CONTENT)
 
+    # Test delete multiple items
+    def test_delete_items_successfully(self):
+        """It should delete multiple items from a shopcart"""
```
Comment (sahishnu) [inline tests/test_routes.py:363]:
```suggestion
        """It should delete multiple items from a shopcart by product_id"""
```

**Review by sahishnu — CHANGES_REQUESTED** [review]
This is great work, just please see the comments

**Comment by kriszhli** [comment]
All comments are resolved!

**Review by sahishnu — APPROVED** [review]
Nice

**Pull request merged**